src/main/java/executor/service/service/impl/proxy/ProxySourceQueue.java

```diff
@@ -39,7 +43,12 @@ public void putProxy(ProxyConfigHolder proxy) {
         try {
             queue.put(proxy);
         } catch (InterruptedException e) {
-            throw new RuntimeException(e);
+            try {
+                log.error("Thread '" + "'" + Thread.currentThread().getName() + " was interrupted");
+                queue.put(proxy);
+            } catch (InterruptedException ex) {
+                throw new RuntimeException(ex);
+            }
         }
```

Comment:
There are no sens, if your queue interrupt it's mean that queue is closed or blocked and it's mean you should to rebuild queue.